test/client/ClientControllerTest.java

```diff
@@ -0,0 +1,47 @@
+package client;
+
+import controllers.ClientController;
+import entities.Client;
+import enums.IocControllers;
+import enums.IocServices;
+import ioc.Ioc;
+import org.junit.Before;
+import org.junit.Test;
+
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.MockitoJUnitRunner;
+import services.implementation.ClientServiceImpl;
+import util.fixtures.ClientServiceFixtures;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Consumer;
+
+
+import static org.junit.Assert.assertEquals;
+import static org.powermock.api.mockito.PowerMockito.when;
+
+public class ClientControllerTest {
+    private ClientController clientController;
+    private ClientServiceImpl clientService;
+    @Before
+    public void setup() {
+        clientService= Mockito.mock(ClientServiceImpl.class);
+        clientController = Ioc.getInstance().get(IocControllers.CLIENT_CONTROLLER);
+    }
+    @Test
+    public void getAll_ShouldReturnListOfClients() {
+        var clients = ClientServiceFixtures.buildClients(3);
+        when(clientService.findAll()).thenReturn(clients);
+        clients.stream().forEach(clientController::add);
+        assertEquals(clientService.findAll(), clientController.getAll());
```

Comment:
This is a bad comparison, service may not return the same thing as the controller

Reply:
I was doing that to check that the service is returning the same thing as the controller. Is that wrong? Is it better to only focus on the client controller methods and not the service?

Reply:
When you are doing unit tests you should care about a single unit

Reply:
I changed it to only focus on the client controller